cloud/amazon/s3.py

```diff
@@ -237,14 +245,23 @@ def upload_s3file(module, s3, bucket, obj, src, expiry, metadata):
     except s3.provider.storage_copy_error, e:
         module.fail_json(msg= str(e))
 
-def download_s3file(module, s3, bucket, obj, dest):
-    try:
-        bucket = s3.lookup(bucket)
-        key = bucket.lookup(obj)
-        key.get_contents_to_filename(dest)
-        module.exit_json(msg="GET operation complete", changed=True)
-    except s3.provider.storage_copy_error, e:
-        module.fail_json(msg= str(e))
+def download_s3file(module, s3, bucket, obj, dest, retries):
+    # retries is the number of loops; range/xrange needs to be one
+    # more to get that count of loops.
+    bucket = s3.lookup(bucket)
+    key = bucket.lookup(obj)
+    for x in xrange(0, retries + 1):
+        try:
+            key.get_contents_to_filename(dest)
+            module.exit_json(msg="GET operation complete", changed=True)
+        except s3.provider.storage_copy_error, e:
+            module.fail_json(msg= str(e))
+        except SSLError as e:
+            # actually fail on last pass through the loop.
+            if x == retries:
```

Comment:
Paranoid programmers would use ```x >= retries``` in case someone in the future changes the looping logic in some unexpected manor ;-)